oshdb-filter/src/main/java/org/heigit/ohsome/oshdb/filter/TagFilterNotEquals.java

```diff
@@ -18,6 +20,11 @@ public OSHDBTag getTag() {
     return this.tag;
   }
 
+  @Override
+  public boolean applyOSH(OSHEntity entity) {
+    return Streams.stream(entity.getVersions()).anyMatch(this::applyOSM);
+  }
+
   @Override
   public boolean applyOSM(OSMEntity entity) {
     return !entity.hasTagValue(tag.getKey(), tag.getValue());
```

Comment:
good point, but let me fix this issue which is unrelated to the feature implemented in this PR in a separate PR :innocent:

Reply:
Makes sense! Would you add here the issue or pr number and resolve this conversation.

Reply:
actually, wait. 

* `applyOSH`: The point of the OSH filter is that it can be run fast and should if at all possible avoid iterating over its versions. The reason is because the `CellIterator` already iterates over those and calls `applyOSM` for them, so when one does this in the `applyOSH` code, one would just unnecessarily duplicate already existing code, and end up with sub-par performance, because other optimizations performed by the `CellIterator` (like the OSMEntity bbox check against the query area of interest) are not applied at that point.
* `applyOSM`: Similar argument, but basically the CellIterator would have already filtered these entities out before calling the function.

Am I overlooking something?